app/js/lib/foxbox/network.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+// Private members.
+const p = {
+  // Private properties.
+  // Whether we can connect to the box via a local connection.
+  local: Symbol('local'),
+  // Whether we can connect to the box via a remote connection.
+  remote: Symbol('remote'),
+  // A reference to the interval to get the online status.
+  pingInterval: Symbol('pingInterval'),
+
+  // Private methods.
+  fetch: Symbol('fetch'),
+  ping: Symbol('ping'),
+  pingBox: Symbol('pingBox')
+};
+
+export default class Network {
+  constructor(settings) {
+    this.settings = settings;
+
+    this[p.local] = false;
+    this[p.remote] = false;
+    this[p.pingInterval] = null;
+  }
+
+  /**
+   * Attach event listeners related to the connection status.
+   *
+   * @return {Promise}
+   */
+  init() {
+    const pingBox = this[p.pingBox].bind(this);
+
+    window.addEventListener('online', pingBox);
+    window.addEventListener('offline', pingBox);
+
+    if ('connection' in navigator && 'onchange' in navigator.connection) {
+      navigator.connection.addEventListener('change', pingBox);
+
+      // We also ping the box every few minutes to make sure it's still there.
+      this[p.pingInterval] = setInterval(pingBox,
+        this.settings.onlineCheckingLongInterval);
+    } else {
+      // If the Network Information API is not implemented, fallback to polling.
+      this[p.pingInterval] = setInterval(pingBox,
+        this.settings.onlineCheckingInterval);
+    }
+
+    this[p.pingBox]();
+
+    return Promise.resolve();
+  }
+
+  get origin() {
+    if (this[p.local]) {
```

Comment:
Because you can be not local and not remote (= disconnected). And if they're both `true`, then you can connect using either the local or remote connection (local is preferred though).
`local` can be understood as 'I can connect to the box using the local network' and `remote` means 'I can connect to the box using the internet'. They're not mutually exclusive.
I thought this implementation was easier when I worked on the logic for `pingBox()`.